pkg/render/certificatemanagement/certificatemanagement.go

```diff
@@ -51,30 +75,38 @@ func (c component) ResolveImages(*operatorv1.ImageSet) error {
 
 func (c component) Objects() (objsToCreate, objsToDelete []client.Object) {
 	if c.cfg.TrustedBundle != nil {
+		// Create the trusted bundle in the namespace that we're installing into.
 		objsToCreate = append(objsToCreate, c.cfg.TrustedBundle.ConfigMap(c.cfg.Namespace))
 	}
+
+	// Iterate each KeyPair and create it where needed. A KeyPair may need to be installed one or more of:
+	// - The "source of truth" namespace, commonly tigera-operator.
+	// - The target namespace that we're installing into.
 	var needsCSRRoleAndBinding bool
 	for _, keyPairCreator := range c.cfg.KeyPairOptions {
 		keyPair := keyPairCreator.keyPair
-		if keyPair != nil {
-			if keyPair.UseCertificateManagement() {
-				if keyPairCreator.renderInOperatorNamespace {
-					objsToDelete = append(objsToDelete, keyPair.Secret(common.OperatorNamespace()))
-				}
-				if keyPairCreator.renderInNamespace {
-					objsToDelete = append(objsToDelete, keyPair.Secret(c.cfg.Namespace))
-				}
-				needsCSRRoleAndBinding = true
-			} else {
-				if keyPairCreator.renderInOperatorNamespace && (!keyPair.BYO() || keyPair.GetName() == certificatemanagement.CASecretName) {
-					objsToCreate = append(objsToCreate, keyPair.Secret(common.OperatorNamespace()))
-				}
-				if keyPairCreator.renderInNamespace {
-					objsToCreate = append(objsToCreate, keyPair.Secret(c.cfg.Namespace))
-				}
+		if keyPair == nil {
```

Comment:
This is just a rephrasing of the previous way of doing this that results in less indentation.